system/modules/news/modules/ModuleNews.php

```diff
@@ -288,7 +288,7 @@ protected function getMetaFields($objArticle)
 	 * @param boolean
 	 * @return string
 	 */
-	protected function generateNewsUrl($objItem, $blnAddArchive=false)
+	public function generateNewsUrl($objItem, $blnAddArchive=false)
```

Comment:
This should be `public static function` otherwise it will trigger a "non-static function cannot be called statically" error.